ndscan/experiment/default_analysis.py

```diff
@@ -413,17 +433,9 @@ def describe_online_analyses(
             if isinstance(v, ResultChannel)
         ]
 
-        analysis_identifier = self.analysis_identifier
```

Comment:
🚨 Potentially contentious... 🚨

If no analysis identifier is given we attempt to generate a unique identifier by mangling together the fit type with the result channels (there is more info we could in principle add here -- even the entire fit description -- but this is pretty good). The generated identifier mangles the names of the result channels fed into the fit (in practice this means that the channels fed into `y` and `y_err` since only 1D fits are supported at present).

Result channels are given a name in the fragment they are originally defined, but can be renamed in the context they are used in (e.g. subscans rename result channels by appending a prefix to avoid collisions). In the original code, the local names within the scan context were used which seems sensible. However, a side effect here is that if the user doesn't explicitly give us an analysis identifier we cannot generate one until `describe_online_analyses`.

This commit introduces a change in behaviour by generating the analysis identifier using the original channel names. The motivation was that (as far as I can tell, but I may be wrong) the result channels need to be defined in the `DefaultAnalysis.__init__` method which does not have access to the scan context. I wanted to generate `p` and `chi2` result channels for the analysis and it seemed sensible to name them based on the analysis identifier, which then required a change in behaviour.

The new behaviour is more likely to cause an accidental collision, but it's not obvious to me how much this is a practical concern (what real use-case would this affect?). So, I think it's fine, but I also haven't got my head fully around the details so may be missing something

NB in many cases the result channels defined in the original fragment will just be the readout `p` and `p_err` so there isn't much value in mangling that into the name since they don't tell us much...

In any case, the tests now pass, but that may say more about the test coverage than the correctness of the design here.

Reply:
In hindsight, a better design here would probably have been to have used an `ExportedResult` to export `chi2` or `p` rather than the implementation here. That way we could revert all of the changes around `analysis_identifier`. It's also a bit more explicit and more consistent with the rest of the interface. It requires a little bit of special casing in `DefaultAnalysis` to handle, but is probably the better design overall. TODO!

nit pick: would `FitResult` be a more descriptive name than `ExportedResult`?